nautobot/core/filters.py

```diff
@@ -760,17 +760,25 @@ def get_filters(cls):
             if filter_name.startswith("_"):
                 del filters[filter_name]
 
-        if "static_groups" not in filters and getattr(cls._meta.model, "is_static_group_associable_model", False):
-            # Add "static_groups" field as the last key
-            from nautobot.extras.models import StaticGroup
-
-            filters["static_groups"] = NaturalKeyOrPKMultipleChoiceFilter(
-                queryset=StaticGroup.objects.all(),
-                field_name="associated_static_groups__static_group",
-                to_field_name="name",
-                query_params={"content_type": cls._meta.model._meta.label_lower},
-                label="Static groups (name or ID)",
-            )
+        if "dynamic_groups" not in filters and getattr(cls._meta.model, "is_dynamic_group_associable_model", False):
+            if not hasattr(cls._meta.model, "static_group_association_set"):
+                logger.warning(
+                    "Model %s has 'is_dynamic_group_associable_model = True' but lacks "
+                    "a 'static_group_association_set' attribute. Perhaps this is due to it inheriting from "
+                    "the deprecated DynamicGroupMixin class instead of the preferred DynamicGroupsModelMixin?",
+                    cls._meta.model,
+                )
+            else:
```

Comment:
Nitpick: Should this be `elif "dynamic_groups" not in filters:` so that this filter can be overridden if desired?

Reply:
Isn't that covered by line 763 above?

Reply:
🤦‍♂️